uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -49,6 +49,9 @@ public static function create($params, $validationMode = self::VALIDATIONS_ON) {
     unset($params['is_deleted']);
 
     $datesChanged = self::datesChanged($params);
+
+    $params = self::resetNonMandatoryDateFieldsIfCalculationUnitChanged($params);
```

Comment:
Just a question, since this is making some changes to the `$params` should validation be called after this?

Reply:
This is not really about validation, but saving the correct value in the database. Depending on the calculation unit, some fields will simply not be present in the `$params` array. For example, when updating a leave request to an Absence Type in hours, the `from_date_type` and `to_date_type` fields will not be sent in the request. For that reason, the validation will pass ([this validation, more specifically](https://github.com/civicrm/civihr/blob/staging/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php#L226-L275)), but the old values will be kept in the database (because they're not included in the `$params` array). This method ensures that the fields will be included in the params array, and that they will be empty, so that they will be set to null in the database.